src/librustc_privacy/lib.rs

```diff
@@ -1321,7 +1321,18 @@ impl<'a, 'tcx, 'v> Visitor<'v> for VisiblePrivateTypesVisitor<'a, 'tcx> {
                     match *trait_ref {
                         None => {
                             for impl_item in impl_items {
-                                visit::walk_impl_item(self, impl_item);
+                                match impl_item.node {
+                                    ast::MethodImplItem(..)
+                                        if self.exported_items.contains(&impl_item.id) ||
```

Comment:
can you maybe add a comment about what's going on here? I'm not sure I understand, though I presume this is related to the rules about privacy and impls?